tests/phpunit/tests/option/option.php

```diff
@@ -524,4 +524,147 @@ public function data_update_option_type_juggling() {
 			array( null, array(), true ),
 		);
 	}
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @covers ::update_option
+	 *
+	 * @dataProvider data_update_option_type_juggling
+	 *
+	 * @param mixed $old_value One of the values to compare.
+	 * @param mixed $new_value The other value to compare.
+	 */
+	public function test_update_option_should_hit_cache_when_loosely_equal_to_existing_value_and_cached_values_are_faithful_to_original_type( $old_value, $new_value ) {
+		add_option( 'foo', $old_value );
+		$num_queries = get_num_queries();
+
+		$updated = update_option( 'foo', $new_value );
+
+		$this->assertFalse( $updated, 'update_option should not return true when values are loosely equal.' );
+		$this->assertSame( $num_queries, get_num_queries(), 'The number of database queries should not change.' );
+	}
```

Comment:
Similar to my other comment, this test is the same as `test_update_loosey_options` from https://core.trac.wordpress.org/changeset/56648, only with an additional assertion from the DB query count. Let's add the assertion to the existing test instead of having this duplicate.